android/src/main/java/dev/matinzd/healthconnect/utils/HealthConnectUtils.kt

```diff
@@ -164,6 +168,14 @@ fun convertDeviceToJSMap(device: Device?): WritableNativeMap? {
   }
 }
 
+fun formatDateKey(instant: Instant): String {
+  val zoneTime = ZonedDateTime.ofInstant(instant, ZoneOffset.systemDefault())
+  val dateKey = zoneTime.format(DateTimeFormatter.BASIC_ISO_DATE)
+
+  // Remove the offset hours
+  return dateKey.substring(0, 8)
```

Comment:
This has probably already been considered however what does the datekey look like when the timezone is +0?
I've seen instances elsewhere where this is returned as just `Z` compared to `+09:00` etc

Reply:
Nice spot thanks! I've updated this to use a custom formatter rather than removing the timezone from the basic format 👍